### PR TITLE
Hw7 task3

### PR DESCRIPTION
**Pull request by E1-Bart0**

### CODE REVIEW

File: All_home_works/hw7/task3.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+def _get_all_lines(board):
+    rows = np.array(board)
+    yield from rows
+    yield from rows.T
+    yield rows.diagonal()
+    yield np.fliplr(rows).diagonal()
+
+
+def _if_all_element_same__add_to(res, line):
```
Comment (Cjkjvfnby) [inline All_home_works/hw7/task3.py]: Let's avoid the mutation of the arguments in this task.

File: All_home_works/hw7/task3.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     [x, o, x]]
+    Return value should be "unfinished"
+    [[-, -, o],
+     [-, o, o],
+     [x, x, x]]
+     Return value should be "x wins!"
+"""
+from typing import List
+
+import numpy as np
```
Comment (Cjkjvfnby) [inline All_home_works/hw7/task3.py]: This is an educational task, let's use pue Python for matrix manipulation.

File: test/hw7/test_task3.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    [
+        ([["-", "-", "o"], ["-", "o", "o"], ["x", "x", "x"]], "x wins!"),
+        ([["o", "o", "o"], ["x", "x", "o"], ["x", "o", "x"]], "o wins!"),
+    ],
+)
+def test_tic_tac_toe_checker__return_win_in_row(input_data, expected):
+    assert expected == tic_tac_toe_checker(input_data)
+
+
+@pytest.mark.parametrize(  # noqa: PT006
```
Comment (Cjkjvfnby) [inline test/hw7/test_task3.py:17]: What is PT006?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: test/hw7/test_task3.py (hunk 3 above)
Comment (E1-Bart0) [reply test/hw7/test_task3.py:17]: https://github.com/E1-Bart0/homework-repo/pull/25#discussion_r627397514

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**